packages/core/options.ts

```diff
@@ -102,31 +102,88 @@ export const fromSnakeCaseWithPrefix: (prefix: string) => KeyFormatSource = (
 	};
 };
 
+// Parameters can be passed in via environment variables, command line or
+// configuration file, and subsequently overwrite the default ones and each other.
+// The passed in values have to be set for externally provided parameters, e.g.
+// CLI parameters, before resolving the final options object.
+// Higher index means higher priority.
+export enum ParameterResolverIndex {
+	DefaultOptions = 1,
+	ConfigurationFile,
+	EnvironmentVariables,
+	CommandLineArguments,
+}
+type ParameterResolver = {
+	name: string;
+	transformKey: KeyFormatSource;
+	failOnUnknown: boolean;
+	object: object;
```

Comment:
What do you think of renaming "object:" into "parameters:"?

Reply:
Sounds good! Changed.